### PR TITLE
Feature/pokemon description

### PR DESCRIPTION
**Pull request by justcier**
What was done:

- Added getPokemonSpecies method and tests for it
- Added flavorText to PokemonDetailsPage
- Created common widget for PokemonNumber and added it to PokemonTileContent

**Screen:**
image|image

### CODE REVIEW

File: lib/common/widgets/pokemon_number.dart [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  Widget build(BuildContext context) {
+    return Text(
+      _showPokemonId(id),
+      style: style,
+    );
+  }
+
+  String _showPokemonId(int pokemonId) {
+    return '#${'O' * (3 - pokemonId.toString().length)}$pokemonId';
+  }
```
Comment (bkkoluch) [inline lib/common/widgets/pokemon_number.dart:23]: Nice, but I feel like it's more of a `parsePokemonId` than `show`, WDYT?

File: lib/features/pokemon_details/presentation/pages/pokemon_details_page.dart [hunk 2]
```diff
@@ -1,64 +1,114 @@
...
+}
+
+class _PokemonDetailsPageState extends State<PokemonDetailsPage> {
+  final PokemonDetailsCubit _pokemonDetailsCubit = getIt<PokemonDetailsCubit>();
+
+  @override
+  void initState() {
+    _pokemonDetailsCubit.getPokemonSpecies(widget.pokemonDetails.id.toString());
+    super.initState();
+  }
```
Comment (bkkoluch) [inline lib/features/pokemon_details/presentation/pages/pokemon_details_page.dart:39]: order, `super.initState()` should always be first (unless there are some exceptions I don't know)

File: lib/features/pokemon_list/presentation/widgets/pokemon_tile_content.dart [hunk 3]
```diff
@@ -45,11 +46,24 @@ class PokemonTileContent extends StatelessWidget {
...
+              FavouriteIcon(
+                pokemonDetails: pokemonDetails,
+              ),
+              PokemonNumber(
+                id: pokemonDetails.id,
+                style: TextStyleTokens.mainTitle
+                    .copyWith(fontSize: Dimensions.sizeL),
+              ),
+            ],
           ),
```
Comment (bkkoluch) [inline lib/features/pokemon_list/presentation/widgets/pokemon_tile_content.dart:67]: If you're only using `padding` parameter then you can use `Padding`

File: test/features/pokemon_details/cubit/pokemon_details_cubit_test.dart [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+          flavorText: tFlavorText,
+        ),
+      ],
+      verify: (_) => [
+        verify(() => getPokemonSpeciesUseCase.call(tId)).called(1),
+      ],
+    );
+
+    blocTest(
+      'should not emit anything if flavorTextEntry == null',
```
Comment (bkkoluch) [inline test/features/pokemon_details/cubit/pokemon_details_cubit_test.dart:45]: It emits something (loaded status)

Comment (bkkoluch) [reply test/features/pokemon_details/cubit/pokemon_details_cubit_test.dart:45]: And the description is wrong as `flavorTextEntries` is empty, not null

File: test/features/pokemon_list/data/data_source/pokemon_list_remote_data_source_impl_test.dart [hunk 5]
```diff
@@ -60,5 +60,45 @@ void main() {
...
+      test(
+        'should return pokemon species',
+        () async {
+          // Arrange
+          when(() => networkService.get(captureAny())).thenAnswer(
+            (_) async => http.Response(
+              fixture(Fixtures.pokemonSpecies),
+              successStatusCode,
+            ),
+          );
```
Comment (bkkoluch) [inline test/features/pokemon_list/data/data_source/pokemon_list_remote_data_source_impl_test.dart:74]: Empty line

File: lib/common/widgets/pokemon_number.dart (hunk 1 above)
Comment (justcier) [reply lib/common/widgets/pokemon_number.dart:23]: You're right, parse sound much better.

File: lib/features/pokemon_details/presentation/pages/pokemon_details_page.dart (hunk 2 above)
Comment (justcier) [reply lib/features/pokemon_details/presentation/pages/pokemon_details_page.dart:39]: Done

File: lib/features/pokemon_list/presentation/widgets/pokemon_tile_content.dart (hunk 3 above)
Comment (justcier) [reply lib/features/pokemon_list/presentation/widgets/pokemon_tile_content.dart:67]: Done

File: test/features/pokemon_details/cubit/pokemon_details_cubit_test.dart (hunk 4 above)
Comment (justcier) [reply test/features/pokemon_details/cubit/pokemon_details_cubit_test.dart:45]: Done

File: test/features/pokemon_list/data/data_source/pokemon_list_remote_data_source_impl_test.dart (hunk 5 above)
Comment (justcier) [reply test/features/pokemon_list/data/data_source/pokemon_list_remote_data_source_impl_test.dart:74]: Done

**Review by bkkoluch — APPROVED** [review]

**Pull request merged**